apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/ApolloException.kt

```diff
@@ -14,3 +14,5 @@ class ApolloHttpException(
     message: String,
     cause: Throwable? = null
 ) : ApolloException(message = message, cause = cause)
+
+class BearerTokenException(message: String, cause: Throwable? = null, val token: String): ApolloException(message = message, cause = cause)
```

Comment:
Would it make sense to somehow extend `ApolloHttpException` here since this is essentially also an `HttpException`

Reply:
The reasoning was to keep the inheritance tree relatively flat. Also now that ApolloException is a sealed class, inheriting from sealed class wouldn't be working.
If the user wants statusCode and headers, they are still readable in `BearerTokenException.cause`

PS: I prefixed with `Apollo`: `ApolloBearerTokenException` to make it consistent with the other classes above.

Reply:
Sounds good. Maybe the type of `cause` can be the concrete type rather than Throwable.

Reply:
But tbh, I'm not sure if that's even necessary since this is not going to be used directly, right?